code/stl/solution/randomize.sol.cpp

```diff
@@ -29,17 +27,27 @@ void compute(int len, T initial, T step) {
     std::vector<T> v(len+1), diffs(len+1);
 
     // fill and randomize v
-    generate(v.begin(), v.end(), Generator<T>(initial, step));
-    shuffle(v.begin(), v.end(), std::default_random_engine{});
+    std::generate(v.begin(), v.end(), Generator<T>(initial, step));
+    // Alternatively:
+    // std::generate(v.begin(), v.end(), [value = initial, step]() mutable {
+    //     const T cur = value;
+    //     value += step;
+    //     return cur;
+    // });
+
+    std::shuffle(v.begin(), v.end(), std::default_random_engine{});
 
     // compute differences
-    adjacent_difference(v.begin(), v.end(), diffs.begin());
+    std::adjacent_difference(v.begin(), v.end(), diffs.begin());
 
     // compute standard deviation of it
-    T sum = reduce(diffs.begin()+1, diffs.end(), T());
-    T sumsq = reduce(diffs.begin()+1, diffs.end(), T(), sumsquare<T>());
-    T mean = sum/len;
-    T variance = sumsq/len - mean*mean;
+    const T sum = std::reduce(diffs.begin()+1, diffs.end(), T());
+    const T sumsq = std::reduce(diffs.begin()+1, diffs.end(), T(), sumsquare<T>());
```

Comment:
```suggestion
    const T sum = std::reduce(diffs.begin(), diffs.end(), T());
    const T sumsq = std::reduce(diffs.begin(), diffs.end(), T(), sumsquare<T>());
```
Why do you skip the first element? Was this meant to be
```c++
    const T sum = std::reduce(diffs.begin(), diffs.end(), diffs[0]);
```
?

Reply:
I think this is correct, since `std::adjacent_difference` only writes differences starting at element 1. So `diffs[0]` is just `v[0]` and `diffs[1]` is the first difference (`v[1] - v[0]`).

Reply:
Sorry, I ignored where those values came from. This is correct, but not very intuitive without any explanation.

Reply:
With your suggestions (that I committed), this is resolved now I guess?